system/src/Grav/Common/Page/Medium/ImageMedium.php

```diff
@@ -72,6 +73,10 @@ public function __construct($items = [], Blueprint $blueprint = null)
         if ($config->get('system.images.cache_all', false)) {
             $this->cache();
         }
+
+        if ($config->get('system.images.watermark.watermark_all', false)) {
+            $this->watermark();
```

Comment:
I'm wondering if we could delay this action as it makes no sense to generate an image unless the user needs it.

Reply:
yeah that doesn't seem to work anyway, i left it there so could ask for some help/advice where it could be added